tests/unit/updateTestResultsFunction.unitTest.ts

```diff
@@ -0,0 +1,78 @@
+import {updateTestResults} from "../../src/functions/updateTestResults";
+import {TestResultsService} from "../../src/services/TestResultsService";
+import {HTTPResponse} from "../../src/models/HTTPResponse";
+import {HTTPError} from "../../src/models/HTTPError";
+
+describe("updateTestResults Function", () => {
+    let event: any;
+    beforeEach(() => {
+        event = {
+            pathParameters: {
+                systemNumber: 1
+            },
+            body: {}
+        };
+    });
+
+    afterEach(() => {
+        event = null;
+    });
+
+    context("when testResult object is not present on the payload", () => {
+        it("should return Error 400 Bad request testResult not provided", async () => {
+            const testResultsMock = jest.fn().mockResolvedValue("Failure");
+            TestResultsService.prototype.updateTestResult = testResultsMock;
+
+            expect.assertions(3);
+            const result = await updateTestResults(event);
+            expect(result).toBeInstanceOf(HTTPResponse);
+            expect(result.statusCode).toEqual(400);
+            expect(result.body).toEqual(JSON.stringify("Bad request testResult not provided"));
+        });
+    });
+
+    context("when msUserDetails object is not present on the payload", () => {
+        it("should return Error 400 Bad request msUserDetails not provided", async () => {
+            const testResultsMock = jest.fn().mockResolvedValue("Failure");
+            TestResultsService.prototype.updateTestResult = testResultsMock;
+            event.body.testResult = "something";
+
+            expect.assertions(3);
+            const result = await updateTestResults(event);
+            expect(result).toBeInstanceOf(HTTPResponse);
+            expect(result.statusCode).toEqual(400);
+            expect(result.body).toEqual(JSON.stringify("Bad request msUserDetails not provided"));
+        });
+    });
+
+    context("Service call fails", () => {
+        it("returns Error", async () => {
+            const myError = new HTTPError(418, "It Broke!");
+            TestResultsService.prototype.updateTestResult = jest.fn().mockRejectedValue(myError);
+            event.body.testResult = "something";
+            event.body.msUserDetails = {msOid: "2", msUser: "dorel"};
+
+            expect.assertions(3);
+            const result = await updateTestResults(event);
+            expect(result).toBeInstanceOf(HTTPResponse);
+            expect(result.statusCode).toEqual(418);
+            expect(result.body).toEqual(JSON.stringify("It Broke!"));
+        });
+    });
+
+    context("Service call succeeds", () => {
+        it("returns 200 + data", async () => {
+            const testResultsMock = jest.fn().mockResolvedValue("Success");
+            TestResultsService.prototype.updateTestResult = testResultsMock;
```

Comment:
Don't we need to reset the mock after the assertions are done?

Reply:
To be honest, I'm not entirely sure, I followed the model of another unit test that was already implemented (getTestResultsByTesterStaffIdFunction) and it doesn't seem to have an impact. 

I noticed though that we were missing the mocking of the service and added that where it was missing.
jest.mock("../../src/services/TestResultsService");